### PR TITLE
RWA-1712

### PR DESCRIPTION
**Pull request by AndreiTrushkov**
### Change description ###
java 17 upgrade

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: config/pmd/ruleset.xml [hunk 1]
```diff
@@ -9,6 +9,8 @@
   <rule ref="category/java/bestpractices.xml">
     <exclude name="GuardLogStatement"/>
     <exclude name="JUnitTestContainsTooManyAsserts"/>
+    <exclude name="LiteralsFirstInComparisons"/>
+    <exclude name="UnusedAssignment"/>
```
Comment (raghera) [inline config/pmd/ruleset.xml:13]: I can't see the errors but can we not fix some of these exclusions?  Unused assignments seems like something easy to get rid of?

File: config/pmd/ruleset.xml [hunk 2]
```diff
@@ -64,6 +68,7 @@
   </rule>
   <rule ref="category/java/errorprone.xml">
     <exclude name="BeanMembersShouldSerialize"/>
+    <exclude name="ReturnEmptyCollectionRatherThanNull"/>
```
Comment (raghera) [inline config/pmd/ruleset.xml:71]: Another one that maybe we could fix?

File: config/pmd/ruleset.xml [hunk 3]
```diff
@@ -41,6 +43,8 @@
     <exclude name="UseUtilityClass"/>
     <exclude name="LoosePackageCoupling"/>
     <exclude name="DataClass"/>
+    <exclude name="CognitiveComplexity"/>
+    <exclude name="CyclomaticComplexity"/>
```
Comment (raghera) [inline config/pmd/ruleset.xml:47]: Do you have an example of code that got flagged by this?
Again in general would be good if we don't turn these off centrally.

File: config/pmd/ruleset.xml (hunk 1 above)
Comment (abhiahirv1) [reply config/pmd/ruleset.xml]: Yes, I rid of it

File: config/pmd/ruleset.xml (hunk 2 above)
Comment (abhiahirv1) [reply config/pmd/ruleset.xml]: Sure, it is removed

File: config/pmd/ruleset.xml (hunk 3 above)
Comment (abhiahirv1) [reply config/pmd/ruleset.xml]: I removed it

**Review by abhiahirv1 — COMMENTED** [review]
Removing extra dependencies

**Review by raghera — APPROVED** [review]

**Pull request merged**